### DISCUSSION TITLE
Show browser-side errors in python console

### DISCUSSION THREAD

**Discussion by mtanco** [h2oai/wave#751] [Ideas]
Today, many Wave-native plot errors will fail silently to both the Wave Server logs and the Python App logs. These are errors like using x_scale as a date in a plot when the data is not in date format. The developer knows something is wrong because the app shows only a white page, but it's unclear what went wrong. 

To find the problem, developers can use `Inspect` in their browser and often find the error reported there. It would be nice if this browser error could be shown in the python logs for easier debugging and for cases when the developer does not have access to the browser but the user could send logs. 

Please let me know if you need example code that fails to make a plot.

**Comment by lo5** [comment]
Yes, we can send the JS crash log over to the Python console, although the errors would be as-is stack traces from [G2](https://github.com/antvis/G2), which might be cryptic in some cases, but still better than nothing. 

A temporary workaround is to open your web browser's Developer Tools and navigate to the console. If a plot crashes, you should find the error message there in red.
